### DISCUSSION TITLE
Fully Customized Web Server

### DISCUSSION THREAD

**Discussion by kyrylobashtenko** [web-infra-dev/modern.js#4226] [General]
Hello, I might have missed your roadmap, but what are the plans for a fully customized web server? Can we expect it anytime soon? Thank you.

**Comment by zllkjc** [comment]
Can you give an example of how you would like to implement this feature?

And we will support fully customized web server. It has been implemented in our internal practice, but has not yet been put into the open source version（Modern.js）.

**Reply by kyrylobashtenko** [reply]
@zllkjc To be honest, I don't have any specific preferences. Something like the example provided in the Next.js documentation on configuring a custom server (https://nextjs.org/docs/pages/building-your-application/configuring/custom-server) seems feasible to me.

In my particular use case, I already have an Express-based app that renders single-spa on the server, and we also have a [node-based ](https://www.keycloak.org/docs/latest/securing_apps/#_nodejs_adapter)Keycloak implementation. So, having a fully customized server might make the migration process to modern.js easier for me.

**Reply by kyrylobashtenko** [reply]
hi, any news about this one? would love to see it!
